README.md

```diff
@@ -1,161 +1,720 @@
+| Name    | Size      | Description                                                                                                     | 
+|---------|-----------| ----------------------------------------------------------------------------------------------------------------| 
+| Boolean | One bit   | A boolean value.                                                                                                |
+| Integer | 64-bit    | A signed integer.                                                                                               |
+| Double  | 64-bit    | [Double-precision] floating-point number, as described in [IEEE 754-2019]. `NaN`s SHOULD NOT be considered valid. |
+| String  | Unbounded | A [UTF8] string. Empty strings MUST be considered valid.                                                        |
+| Bytes   | Unbounded | Zero or more binary octets.                                                                                     |
+| CID     | Unbounded | A [CID]. Avoiding the identity hash is RECOMMENDED.                                                             |
 
-Since content addressing schemes are backed by cryptographically secure hash functions, their use here prevents forgery of IDs by attackers, and guarantees that CID-based dependencies between tuples will be acyclic.
+Note that there is no primitive `unit` or `null`. Emulating these constructs in data modeling is NOT RECOMMENDED, as they are semantically anemic and alternative modelings are nearly always available. For example, a common use case for using `null` is unset a field. CRDTs SHOULD use history-preserving tombstoning for this purpose instead.
 
-These properties are further leveraged in the design and use of byzantine-fault tolerant CRDTs, as described in [CRDTs](pomo_db/CRDTs.md).
+### 4.1.2 Fact
 
-TODO: Update CRDT link once that info is described somewhere
+The extensional database MUST only store facts as quads (4-tuples) in EAVC[^spoc] format. All fields are REQUIRED, though the causal set MAY be empty.
 
-## 2.5 Provenance Tracking
+[^spoc]: This is [sometimes called SPOC][Hexastore] (🖖) for ["subject-predicate-object-context"][Named Graphs]
 
-TODO: [See notes](https://discord.com/channels/478735028319158273/1033502043656171561/1035339517021921280)
+| Field         | Type          | Description                                        |
+|---------------|---------------|----------------------------------------------------|
+| **E**ntity ID | `Bytes`       | [Entity ID]                                        |
+| **A**ttribute | [`Attribute`] | The name of the value's relationship to the entity |
+| **V**alue     | [`Value`]     | The (primitive) value being associated             |
+| **C**auses    | `Set CID`     | Any direct causal links                            |
 
-## 2.6 Query Engine
+``` typescript
+type Fact = {
+  eid: Bytes;
+  attr: Attribute;
+  val: Value;
+  causal: Set<Cid>;
```

Comment:
Were you not thinking of `causal` as a set of key-value pairs, naming the CIDs? That's how the current implementation works, stemming from an earlier discussion we had about how to capture semantically different links from the same tuple, in a BFT way, though I'm not actually aware of any designs that require that functionality.

Reply:
Chatted about this on a live call. TL;DR we're using a set like given here 👍